app/models/label/LabelTable.scala

```diff
@@ -503,19 +538,59 @@ object LabelTable {
 
   /**
     * Retrieves a list of labels to be validated
-    * @param count  Length of list
-    * @return       Seq[LabelValidationMetadata]
+    * @param count        Length of list
+    * @param labelTypeId  Label Type of each label in the list
+    * @return             Seq[LabelValidationMetadata]
     */
-  def retrieveRandomLabelListForValidation(count: Int) : Seq[LabelValidationMetadata] = db.withSession { implicit session =>
+  def retrieveLabelListForValidation(userId: UUID, count: Int, labelType: Int) : Seq[LabelValidationMetadata] = db.withSession { implicit session =>
     var labelList = new ListBuffer[LabelValidationMetadata]()
+    var labelIdList = new ListBuffer[Int]()
     for (a <- 1 to count) {
-      labelList += retrieveSingleRandomLabelForValidation()
+      labelList += retrieveSingleRandomLabelFromLabelTypeForValidation(userId, labelType, Some(labelIdList))
     }
     val labelSeq: Seq[LabelValidationMetadata] = labelList
     labelSeq
   }
 
   /**
+    * Retrieve a list of labels for validation with a random label id
+    * @param count  Number of labels in the list
+    * @return       Seq[LabelValidationMetadata]
+    */
+  def retrieveRandomLabelListForValidation(userId: UUID, count: Int) : Seq[LabelValidationMetadata] = db.withSession { implicit session =>
+    // We are currently assigning label types to missions randomly.
+    val labelTypeId: Int = retrieveRandomValidationLabelTypeId()
+    val labelSeq: Seq[LabelValidationMetadata] = retrieveLabelListForValidation(userId, count, labelTypeId)
+    labelSeq
```

Comment:
I wouldn't bother putting this on two separate lines. I don't think it adds anything. I know I see it elsewhere in the code, but I don't see why :sweat_smile:
`retrieveLabelListForValidation(userId, count, labelTypeId)` or
`return retrieveLabelListForValidation(userId, count, labelTypeId)` seem sufficient to me.